projects/step-frontend/src/lib/modules/timeseries/performance-view/filter-bar/dashboard-filter-bar.component.html

```diff
@@ -0,0 +1,114 @@
+<div class="time-container">
+  <div class="ranger-section" [class.compact]="compactView">
+    <step-execution-time-selection [context]="context"></step-execution-time-selection>
+  </div>
+  <div>
+    <step-time-range-picker
+      (selectionChange)="handleTimeRangeChange($event)"
+      [selectOptions]="timeRangeOptions"
+      [activeSelection]="activeTimeRange"
+      [compact]="compactView"
+    >
+    </step-time-range-picker>
+  </div>
+</div>
+
+<div class="filters-section">
+  <step-alert type="warning" class="warning" *ngIf="rawMeasurementsModeActive">
+    You are using custom filters which may slow down the dashboard.
+  </step-alert>
+
+  <div class="settings-section" *ngIf="compactView">
+    <mat-button-toggle-group [(ngModel)]="oqlModeActive" (change)="toggleOQLMode()">
+      <mat-button-toggle [value]="false"> Basic</mat-button-toggle>
+      <mat-button-toggle [value]="true"> OQL</mat-button-toggle>
+    </mat-button-toggle-group>
+    <step-timeseries-grouping
+      [dimensions]="activeGrouping"
+      (groupingChange)="handleGroupingChange($event)"
+    ></step-timeseries-grouping>
+  </div>
+
+  <div class="filters">
+    <mat-menu #addFilterMenu="matMenu">
+      <ng-container *ngFor="let option of filterOptions">
+        <button mat-menu-item *ngIf="!option.isHidden" (click)="addFilterItem(option)">
+          <span>{{ option.label }}</span>
+        </button>
+      </ng-container>
+      <button mat-menu-item [matMenuTriggerFor]="typesMenu">Other</button>
+      <button mat-menu-item (click)="openDiscovery()">Discover...</button>
+    </mat-menu>
+
+    <mat-menu #typesMenu="matMenu">
+      <button mat-menu-item (click)="addCustomFilter('FREE_TEXT')">Text Filter</button>
+      <button mat-menu-item (click)="addCustomFilter('NUMERIC')">Numeric Filter</button>
+      <button mat-menu-item (click)="addCustomFilter('DATE')">Date Filter</button>
+    </mat-menu>
+
+    <div
+      class="filers-row"
+      [style]="oqlModeActive ? 'overflow: unset' : ''"
+      [style.flexWrap]="compactView ? 'wrap' : 'nowrap'"
+    >
+      <ng-container *ngIf="!oqlModeActive">
+        <ng-container *ngFor="let item of _internalFilters; let i = index">
```

Comment:
Add trackBy

Reply:
@dvladir I think track by identity is correct here. there is no unique identifier on these items